stl/inc/chrono

```diff
@@ -2105,8 +2105,9 @@ namespace chrono {
 
     template <class _Ty>
     _NODISCARD const _Ty* _Locate_zone_impl(const vector<_Ty>& _Vec, string_view _Name) {
-        const auto _Result = _STD find_if(_Vec.begin(), _Vec.end(), [&](auto& _Tz) { return _Tz.name() == _Name; });
-        return _Result == _Vec.end() ? nullptr : &*_Result;
+        const auto _Result = _STD lower_bound(
+            _Vec.begin(), _Vec.end(), _Name, [](const auto& _Tz, const auto& _Sv) { return _Tz.name() < _Sv; });
+        return (_Result == _Vec.end() || _Result->name() != _Name) ? nullptr : &*_Result;

```

Comment:
This is partially pre-existing, but we try to avoid complicated conditional (ternary) operators in new code - they pack too much into a single line, making it harder to see what's happening. As the condition for this one is expanding, we should just use an `if`-statement here.

Additionally, while the positive/negative nature of the original condition didn't really matter, now that it's expanded I believe that it would be clearer to phrase it positively: "if we found a result and the name matches, then success".